src/App.js

```diff
@@ -0,0 +1,198 @@
+import { useState, useEffect } from "react";
+import NewBoardForm from "./components/NewBoardForm";
+import NewCardForm from "./components/NewCardForm";
+import BoardList from "./components/BoardList";
+import CardList from "./components/CardList";
+import "./styles/App.css";
+import {
+  postBoardAsync,
+  postCardAsync,
+  likeCardAsync,
+  deleteCardAsync,
+  selectBoardAsync,
+  getAllBoardsAsync,
+} from "./apiCalls";
+
+function App() {
+  const [boardData, setBoardData] = useState([]);
+  const [selectedBoardId, setSelectedBoardId] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const displayAllBoards = () => {
+    getAllBoardsAsync()
+      .then((boards) => {
+        setBoardData(boards);
+      })
+      .catch((err) => {
+        console.log(err);
+        setErrorMessage("error displaying all boards");
+      });
+  };
+
+  useEffect(() => {
+    displayAllBoards();
+  }, [boardData]);
```

Comment:
There's an issue here folks will want to watch out for on future projects. If you go to the front end link and open up the dev tools to the network tab, you'll see the page is caught in an infinite loop of making calls to fetch all board data. For folks on mobile or with a limited data plan, a site that does this can eat away at data, especially if what's being fetched is significantly larger, like images or videos.

The loop is being caused by the `useEffect` statement: by placing `boardData` in the dependencies list on line 34, any time the state `boardData` is updated, the `useEffect` block will be ran. This block calls `displayAllBoards()` which fetches and updates the `boardData` state...which triggers the `useEffect` block to run, and this goes on in an infinite loop.

If the goal is to fetch the all the boards data once when we land on the screen, we would want to remove `boardData` from line 34.